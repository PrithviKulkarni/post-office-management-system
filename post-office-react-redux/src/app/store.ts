import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import letterReducer from '../features/LetterSlice';
import parcelReducer from '../features/ParcelSlice';
import depotReducer from '../features/DepotSlice';
import driverReducer from '../features/DriverSlice' 
export const store = configureStore({
  reducer: {
    letter: letterReducer,
    parcel: parcelReducer,
    depot: depotReducer,
    driver: driverReducer
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
