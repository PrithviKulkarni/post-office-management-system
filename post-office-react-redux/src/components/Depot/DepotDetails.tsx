import { useSelector } from 'react-redux'
import { useParams } from 'react-router-dom'
import { RootState } from '../../app/store'
import { getDepotById } from '../../features/DepotSlice'

const DepotDetails = () => {
  let { depId } = useParams()

  let depotId: string = '0'

  if (!depId) {
    depId = '0'
  }
  depotId = depId

  const depots = useSelector((state: RootState) => getDepotById(state, depotId))

  if (!depots) {
    return (
      <>
        <section data-testid="depotDetails" ></section>
      </>
    )
  } else {
    return (
      <div data-testid="depotDetails">
        <div className='card'>
          <h1>{depots.city} Depot</h1>
          <div className='card-header'>Special Services</div>
          <ul className='list-group list-group-flush'>
            {depots?.services.map(service => (
              <li className='list-group-item' key={service.service}>
                {' '}
                Service: {service.service}
              </li>
            ))}
          </ul>
        </div>
      </div>
    )
  }
}

export default DepotDetails
