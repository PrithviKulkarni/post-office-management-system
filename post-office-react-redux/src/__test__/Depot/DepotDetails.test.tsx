import { render, cleanup } from '@testing-library/react';
import DepotDetails from '../../components/Depot/DepotDetails';
import { screen } from '@testing-library/react';
import { Provider } from "react-redux";
import { store } from '../../app/store';

afterEach(cleanup);
it("Depot's details render test", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <DepotDetails />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("Verifies depot's details has rendered", () => {
  render(
    <Provider store={store}>
      <DepotDetails />
    </Provider>
  );
  
  expect(screen.getByTestId("depotDetails")).toBeInTheDocument();
});