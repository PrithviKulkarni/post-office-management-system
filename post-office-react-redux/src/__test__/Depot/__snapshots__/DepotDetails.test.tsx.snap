// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Depot's details render test 1`] = `
<DocumentFragment>
  <section
    data-testid="depotDetails"
  />
</DocumentFragment>
`;
